model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -47,4 +51,15 @@ default void checkName() {
   static Branch of(String name, String hash) {
     return builder().name(name).hash(hash).build();
   }
+
+  /**
+   * Returns a map of metadata properties that describe this branch. Note that these properties
+   * <b>can be added</b> by the server when a {@link Branch} instance is returned.
+   *
+   * @return A map of metadata properties that describe this branch. Note that these properties
+   *     <b>can be added</b> by the server when a {@link Branch} instance is returned.
+   */
+  @Nullable
+  @JsonInclude(Include.NON_NULL)
+  Map<String, Object> metadataProperties();
```

Comment:
Why not have a strongly typed holder class for the metadata? It could extend `Map` if we want to allow clients to write more generic code for processing those properties.

Reply:
what would be the benefit? To me it seemed having a simple `Map<String, Object>` would be flexible enough in case we want to put more info in the future